docs/DeveloperGuide.md

```diff
@@ -290,30 +290,277 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  * 2a1. LMS requests for the correct data.
+  * 2a2. User enters new data.
+  * Steps 2a1-2a2 are repeated until the data entered are correct.
+  * Use case resumes from step 3.
+
+* \*a. At any time, User chooses to cancel the cancellation process.
+  * \*a1. LMS requests to confirm the cancellation of the process.
+  * \*a2. User confirms the cancellation of the process.
+  * Use case ends.
+
+---
+
+#### Use case: Schedule a make-up lesson
+
+**MSS**
+
+1. User enters command to schedule a make-up lesson.
+2. LMS parses the make-up lesson details from the user's command.
+3. LMS requests for confirmation.
+4. User confirms the scheduling.
+5. LMS schedules the make-up lesson and notifies the user.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. LMS detects an error in the entered data.
+  * 2a1. LMS requests for the correct data.
+  * 2a2. User enters new data.
+  * Steps 2a1-2a2 are repeated until the data entered are correct.
+  * Use case resumes from step 3.
+
+* \*a. At any time, User chooses to cancel the scheduling process.
+  * \*a1. LMS requests to confirm the cancellation of the process.
+  * \*a2. User confirms the cancellation of the process.
+  * Use case ends.
+
+---
+
+#### Use case: Delete a student
+
+**MSS**
+
+1. User enters command to delete a student.
+2. LMS parses the student details from the user's command.
+3. LMS requests for confirmation.
+4. User confirms the deletion.
+5. LMS deletes the student from the database and notifies the user.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. LMS detects an error in the entered data.
+  * 2a1. LMS requests for the correct data.
+  * 2a2. User enters new data.
+  * Steps 2a1-2a2 are repeated until the data entered are correct.
+  * Use case resumes from step 3.
+
+* \*a. At any time, User chooses to cancel the deletion process.
+  * \*a1. LMS requests to confirm the cancellation of the process.
+  * \*a2. User confirms the cancellation of the process.
+  * Use case ends.
+
+---
+
+#### Use case: View the list of commands
+
+**MSS**
+
+1. User enters command to view the list of commands.
+2. LMS retrieves the list of available commands.
+3. LMS displays the list of commands to the user.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. LMS detects an error while retrieving the list.
+  * 2a1. LMS shows an error message.
+  * Use case ends.
+
+---
+
+#### Use case: Save and retrieve data after application restart
+
+**MSS**
+
+1. User enters data into the application.
+2. User chooses to save the data.
+3. LMS saves the data to persistent storage.
+4. User turns off the application.
+5. User restarts the application.
+6. LMS retrieves the data from persistent storage and displays it to the user.
+
+   Use case ends.
+
+**Extensions**
+
+* 3a. LMS detects an error while saving the data.
+  * 3a1. LMS shows an error message.
+  * Use case ends.
+
+* 6a. LMS detects an error while retrieving the data.
+  * 6a1. LMS shows an error message.
+  * Use case ends.
```

Comment:
This should not be here, because saving data and loading data is done automatically.

Reply:
I personally think saving and retrieving data should be in use cases. It's an interaction between user and system that user expects as granted

Reply:
Use case should display only externally visible behaviour, and should involve interactions between user and system. Saving data is done solely by the system, there is no user interaction involve. Additionally, saving data is done behind the scenes by the system, with no response to the user.